crowd_anki/importer/import_dialog.py

```diff
@@ -0,0 +1,189 @@
+from collections import namedtuple
+from dataclasses import dataclass, field
+from enum import Enum
+
+from aqt.qt import *
+from typing import List, Tuple, Dict
+
+from .import_ui import Ui_Dialog as ConfigUI
+from ..config.config_settings import ConfigSettings
+from ..utils.constants import UUID_FIELD_NAME
+
+from ..utils.utils import string_cs_to_list
+
+
+@dataclass
+class ConfigEntry:
+    config_name: str
+    default_value: any
+
+
+@dataclass
+class PersonalFieldsHolder:
+    personal_fields: Dict[str, List[str]] = field(init=False, default_factory=lambda: dict())
+
+    def has_pf(self, field_name, *keys):
+        def _check_pfs(key):
+            if key in self.personal_fields:
+                if field_name in self.personal_fields[key]:
+                    return True
+            return False
+        return any(_check_pfs(key) for key in keys)
+
+    def add_field(self, model_name, model_id, field_name):
+        def _add(key):
+            if key not in self.personal_fields:
+                self.personal_fields.setdefault(key, [field_name])
+            else:
+                self.personal_fields[key].append(field_name)
+        _add(model_id)
+        _add(model_name)
+
+
+@dataclass
+class ImportDefaults(PersonalFieldsHolder):
+    import_message: str = field(init=False)
+    suggest_tag_imported_cards: bool = field(init=False)
+
+    class Properties(Enum):
+        IMPORT_MESSAGE = ConfigEntry("import_message", "")
+        SUGGEST_TAG_IMPORTED_CARDS = ConfigEntry("suggest_tag_imported_cards", False)
+
+    @classmethod
+    def from_dict(cls, settings_dict: dict) -> 'ImportDefaults':
+        new_cls = cls()
+        for prop in cls.Properties:
+            setattr(new_cls, prop.value.config_name, settings_dict.get(prop.value.config_name, prop.value.default_value))
+        new_cls._setup_personal_fields(settings_dict)
+        return new_cls
+
+    def _setup_personal_fields(self, settings_dict):
+        models = settings_dict.get("note_models", [])
+        for model_name_or_id in models:
+            self.personal_fields.setdefault(model_name_or_id, models[model_name_or_id].get("personal_fields", []))
+
+
+@dataclass
+class ImportConfig(PersonalFieldsHolder):
+    add_tag_to_cards: List[str]
+
+    use_header: bool
```

Comment:
btw what is header?

Reply:
>I gave implementing Use Headers (You call this the Deck and Metadata) a go, but since it is split up into two segments, which are quite interlinked to all the others, I gave up for now. Also if the user has unticked Use Headers then perhaps they should also be forced to choose an existing deck for the new cards to go in 🤔 or just put them all in Default? What do you think? 🤔

ah, you mention that it's not implemented for now. I believe it should be removed until it is. I'm still confused on what do you mean by header though?

Reply:
I'm not sure if it's useful to combine all those things together. I find it confusing, I think I won't be alone.   
Deck options can be separated out and it makes sense to do so #31   
But CrowdAnki is build on the notion of the Deck, I'd say it wont be straightforward to diverse it from the notion of a deck and I'm not sure if it makes sense to do so either

Reply:
See reply here https://github.com/Stvad/CrowdAnki/issues/31#issuecomment-653851949 :wink: